### PR TITLE
554 loading animation onscroll + mobx

### PR DESCRIPTION
**Pull request by alex-cory**

### CODE REVIEW

File: src/scenes/AllEvents/index.js [hunk 1]
```diff
@@ -1,88 +1,46 @@
...
+          loadMore={loadMoreEvents}
+          loadingMore={loadingMore}
         />
-      </div>
+      </Fragment>
     );
   }
 }
+
+const Loading = () => <Row><_Loading style={{ width: '5rem', height: '5rem', marginTop: 360 }} text="Loading..." /></Row>;
```
Comment (amazingandyyy) [inline src/scenes/AllEvents/index.js]: cn: `加载中...`
kr: `로딩...`

File: src/components/InfiniteScroll/index.js [hunk 2]
```diff
@@ -40,7 +43,23 @@ export default class InfiniteScroll extends Component {
     return (
       <Grid container spacing={this.props.spacing}>
         {this.props.data}
+        {this.props.loadingMore && <LoadingMore />}
       </Grid>
     );
   }
 }
+
+const LoadingMore = () => <Row><Loading text="Loading More..." /></Row>;
```
Comment (amazingandyyy) [inline src/components/InfiniteScroll/index.js]: cn: `加载更多...`
kr: `로딩...`

File: src/stores/models/Oracle.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+export default class Oracle {
+  phase = ''
+  status = ''
+  txid = ''
+  address = ''
+  amounts = []
+  blockNum
+  consensusThreshold = '10000000000'
```
Comment (dwalintukan) [inline src/stores/models/Oracle.js]: we shouldnt hardcode this. this can change on the fly.

File: src/stores/models/Oracle.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+import { OracleStatus, TransactionType, TransactionStatus, Phases } from '../../constants';
+import { satoshiToDecimal } from '../../helpers/utility';
+const { Pending } = TransactionStatus;
+const { bet, vote, setResult, finalize, withdraw } = Phases;
+
+const messages = defineMessages({
+  placeBet: { id: 'bottomButtonText.placeBet', defaultMessage: 'Place Bet' },
+  setResult: { id: 'str.setResult', defaultMessage: 'Set Result' },
+  arbitrate: { id: 'bottomButtonText.arbitrate', defaultMessage: 'Arbitrate' },
+  finalizeResult: { id: 'str.finalizeResult', defaultMessage: 'Finalize Result' },
```
Comment (dwalintukan) [inline src/stores/models/Oracle.js]: need to define message of `pending`. let me merge my PR first then can rebase and add here.

File: src/stores/models/Oracle.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+ * @param {oracle} oracle
+ */
+const getPhase = ({ token, status }) => {
+  const [BOT, QTUM] = [token === 'BOT', token === 'QTUM'];
+  if (QTUM && ['VOTING', 'CREATED'].includes(status)) return bet;
+  if (BOT && status === 'VOTING') return vote;
+  if (QTUM && ['WAITRESULT', 'OPENRESULTSET'].includes(status)) return setResult;
+  if (BOT && status === 'WAITRESULT') return finalize;
+  if (((BOT || QTUM) && status === 'WITHDRAW') || (QTUM && status === 'PENDING')) return withdraw;
+  throw Error(`Invalid Phase determined by these -> TOKEN: ${token} STATUS: ${status}`);
```
Comment (dwalintukan) [inline src/stores/models/Oracle.js]: this can be copied over from my PR changes.

File: src/stores/models/Oracle.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    this.phase = getPhase(oracle);
+    this.amounts = oracle.amounts.map(satoshiToDecimal);
+    this.consensusThreshold = satoshiToDecimal(oracle.consensusThreshold);
+
+    const { ApproveSetResult, SetResult, ApproveVote, Vote, FinalizeResult, Bet } = TransactionType;
+    const pendingTypes = {
+      bet: [Bet],
+      vote: [ApproveVote, Vote],
+      setResult: [ApproveSetResult, SetResult],
+      finalize: [FinalizeResult],
```
Comment (dwalintukan) [inline src/stores/models/Oracle.js]: these can be copied from my PR

File: src/stores/models/Oracle.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+    this.unconfirmed = (!oracle.topicAddress && !oracle.address) || isPending;
+
+    this.isUpcoming = this.phase === vote && oracle.status === OracleStatus.WaitResult;
+
+    this.buttonText = {
+      bet: messages.placeBet,
+      setResult: messages.setResult,
+      vote: messages.arbitrate,
+      finalize: messages.finalizeResult,
+      withdraw: messages.withdraw,
```
Comment (dwalintukan) [inline src/stores/models/Oracle.js]: these can be copied over

File: src/scenes/App/globalHub.js [hunk 8]
```diff
@@ -1,5 +1,7 @@
+/* eslint-disable */
```
[the remark this answers is not in the document]
Comment (alex-cory) [reply src/scenes/App/globalHub.js]: I removed this already. Was an accident 😅

**Comment by amazingandyyy** [comment]
I just approved it in case I am sleeping when you guys shipping.

**Comment by alex-cory** [comment]
https://github.com/bodhiproject/bodhi-ui/issues/554

**Pull request merged**